### PR TITLE
Mobile menu javascript

### PR DESCRIPTION
**Pull request by principles31**
**Portfolio: Mobile Menu**
> The goal of this project is to implement the mobile menu using javascript.

> The following features have been implemented in the project.

> When the user clicks (or taps) the hamburger button, the mobile menu appears.
> When the user clicks (or taps) the close (X) button, the mobile menu disappears.
> When the user clicks (or taps) any of the mobile menu options, the mobile menu disappears.
> When the user clicks (or taps) any of the mobile menu options, a correct part of the page is displayed.

> Template 5 was used.
>  It has been tested in my browser and working perfectly.
>  The issue with the Humbarger mentioned in the previous review has been solved.
> Below is a screenshot of the project 
** Mobile Menu Section**
menu_picture

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -1,276 +1,249 @@
...
+          <li class=""><a href="#about">About</a></li>
+          <li class=""><a href="#contact">Contact</a></li>
+        </ul>
+
+        <ul>
+          <li class="li_first"><a href="#hello" id="hello">Hello</a></li>
+          <li class=""><a href="#portfolio">Portfolio</a></li>
+          <li class=""><a href="#about">About</a></li>
+          <li class=""><a href="#contact">Contact</a></li>
+        </ul>
```
Comment (ShadyShawkat) [inline index.html]: - [ ] Kindly make sure that the modal takes the entirety of the screen's width and height and the user can't scroll to the page's content when the modal is opened.
image

Comment (ShadyShawkat) [inline index.html]: - [ ] Please make sure that the mobile hamburger menu button only appears on the mobile view and not the desktop view. Currently, the hamburger menu is displayed on both the mobile and desktop views.
image

File: index.html [hunk 2]
```diff
@@ -1,276 +1,249 @@
...
+        <div class="mobile-icons">
+          <i class="fa fa-bars fa-xl hamb"></i>
+          <i class="fa fa-xmark fa-solid fa-xl close-icon"></i>
+        </div>
+        <img src="./img/img-before.svg" class="img-before" alt="">
+        <ul class="mobile-options">
+          <li class="li_first"><a href="#hello">Hello</a></li>
+          <li class=""><a href="#portfolio">Portfolio</a></li>
+          <li class=""><a href="#about">About</a></li>
+          <li class=""><a href="#contact">Contact</a></li>
```
Comment (ShadyShawkat) [inline index.html]: - [ ] Please be sure that the modal is closed and hidden after the user clicks any nav link or the `x` button. Currently, the modal isn't closed or hidden after the user clicks any nav link.
- [ ] Kindly make sure that after the modal is closed through nav links or the `x` button, the page returns to the original design, and only the modal is hidden out of sight. Currently, after the modal is closed, the following behavior occurs. (See image below)

Before opening the modal:
image
After closing the modal:
image

**Review by ShadyShawkat — CHANGES_REQUESTED** [review]
## Hi Team, 

Good job so far! 👍🏻 
There are some issues that you still need to work on to go to the next project but you are almost there!

Image

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by CollinsTatang — CHANGES_REQUESTED** [review]
Hi Team, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️

- [x] The hamburger menu is not stable and I can't find the closing `X` bottom.
### **Take a look at the mobile view**

https://user-images.githubusercontent.com/55596970/163474599-e26d101e-dea8-4199-86bd-1c6016bf340c.mp4




_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @CollinsTatang in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by principles31** [comment]
@CollinsTatang, thank you for the feedback. 
function showmodal() {
  modal.style.display = 'block';
  flexUl.style.display = 'block';
}

btn.addEventListener('click', showmodal);

function hideModal() {
  modal.style.display = 'none';
  flexUl.style.display = 'none';
}

flexUl.addEventListener('click', hideModal);
span.addEventListener('click', hideModal);

please kindly help which function best for implementing the above issues hence ....function.addEventListener is rejected by the javascript ESlint.

**Comment by CollinsTatang** [comment]
> @CollinsTatang, thank you for the feedback. function showmodal() { modal.style.display = 'block'; flexUl.style.display = 'block'; }
> 
> btn.addEventListener('click', showmodal);
> 
> function hideModal() { modal.style.display = 'none'; flexUl.style.display = 'none'; }
> 
> flexUl.addEventListener('click', hideModal); span.addEventListener('click', hideModal);
> 
> please kindly help which function best for implementing the above issues hence ....function.addEventListener is rejected by the javascript ESlint.

Hello @principles31 
Can we have a call?

**Comment by principles31** [comment]
@CollinsTatang Yes  please, I'm available

**Comment by CollinsTatang** [comment]
> 

OK

**Comment by principles31** [comment]
@CollinsTatang It has been resolved now. Thank you for your effort to improve my code. the best reviewer so far ...keep it up

**Comment by CollinsTatang** [comment]
Thanks 
All the best as you move to the next project

**Comment by CollinsTatang** [comment]
> @CollinsTatang It has been resolved now. Thank you for your effort to improve my code. the best reviewer so far ...keep it up

Thanks
All the best as you move to the next project

**Review by AnselemOdims — APPROVED** [review]
### Hi  @principles31 🙋‍♀️, 
Good job in making the required changes 👏 

# STATUS APPROVED ✔
Your project is complete!  There is nothing else to say other than... it's time to merge it 🔗 🚢 
Congratulations! 🎉 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers 🥂  and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Use the tag @AnselemOdims for a quicker reply


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by principles31** [comment]
@AnselemOdims @CollinsTatang @ShadyShawkat.Thank you for your approval. it takes a lot to make this project work and you are part of that success and the previous reviewers.

**Pull request merged**